utils/ziputil/zip_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018 the Service Broker Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ziputil
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/gcp-service-broker/utils/stream"
+)
+
+var exampleFile = []byte{
+	0x50, 0x4b, 0x3, 0x4, 0x14, 0x0, 0x8, 0x0, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x13, 0x0,
+	0x0, 0x0, 0x70, 0x61, 0x74, 0x68, 0x2f, 0x74, 0x6f, 0x2f, 0x6d, 0x79,
+	0x2f, 0x66, 0x69, 0x6c, 0x65, 0x2e, 0x74, 0x78, 0x74, 0xf2, 0x48, 0xcd,
+	0xc9, 0xc9, 0xd7, 0x51, 0x28, 0xcf, 0x2f, 0xca, 0x49, 0x51, 0x4, 0x4, 0x0,
+	0x0, 0xff, 0xff, 0x50, 0x4b, 0x7, 0x8, 0xe6, 0xc6, 0xe6, 0xeb, 0x13, 0x0,
+	0x0, 0x0, 0xd, 0x0, 0x0, 0x0, 0x50, 0x4b, 0x1, 0x2, 0x14, 0x0, 0x14, 0x0,
+	0x8, 0x0, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0xe6, 0xc6, 0xe6, 0xeb, 0x13, 0x0,
+	0x0, 0x0, 0xd, 0x0, 0x0, 0x0, 0x13, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x70, 0x61, 0x74, 0x68, 0x2f,
+	0x74, 0x6f, 0x2f, 0x6d, 0x79, 0x2f, 0x66, 0x69, 0x6c, 0x65, 0x2e, 0x74, 0x78,
+	0x74, 0x50, 0x4b, 0x5, 0x6, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, 0x41,
+	0x0, 0x0, 0x0, 0x54, 0x0, 0x0, 0x0, 0x0, 0x0}
+
+func openZipfile() *zip.Reader {
+	r, _ := zip.NewReader(bytes.NewReader(exampleFile), int64(len(exampleFile)))
+	return r
+}
+
+func ExampleList() {
+	zf := openZipfile()
+	List(zf, os.Stdout)
+
+	// Output: MODE        SIZE  NAME
+	// -rw-rw-rw-  13    path/to/my/file.txt
+}
+
+func ExampleFind() {
+	zf := openZipfile()
+	fmt.Println(Find(zf, "does", "not", "exist"))
+	fmt.Println(Find(zf, "path/to/my", "file.txt").Name)
+
+	// Output: <nil>
+	// path/to/my/file.txt
+}
+
+func ExampleOpen() {
+	zf := openZipfile()
+	source := stream.FromReadCloserError(Open(zf, "path/to/my/file.txt"))
+	stream.Copy(source, stream.ToWriter(os.Stdout))
+
+	// Output: Hello, world!
+}
+
+func ExampleExtract() {
+	tmp, _ := ioutil.TempDir("", "ziptest")
```

Comment:
Again recommend not swallowing the error here, instead consider creating a test factory which creates tempdirs and handles the error for you.